### ISSUE TITLE
Detect invocation from outside of Rust project

### ISSUE THREAD

**Issue by regexident** [pegasos1/cargo-profiler#19]
If you try to run, say `cargo profiler callgrind --release` from a directory that's not a cargo project root (as in "does not contain a `Cargo.toml` file") you get an exception:

```
thread '<main>' panicked at 'Error in encoding manifest string into JSON: SyntaxError("EOF While parsing value", 1, 1)', ../src/libcore/result.rs:785
stack backtrace:
   1:        0x10b3f4c18 - std::sys::backtrace::tracing::imp::write::h9fb600083204ae7f
   2:        0x10b3f82a5 - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hca543c34f11229ac
   3:        0x10b3f7ede - std::panicking::default_hook::hc2c969e7453d080c
   4:        0x10b3e8de0 - std::panicking::rust_panic_with_hook::hfe203e3083c2b544
   5:        0x10b3f8866 - std::panicking::begin_panic::h4889569716505182
   6:        0x10b3e9a28 - std::panicking::begin_panic_fmt::h484cd47786497f03
   7:        0x10b3f84bf - rust_begin_unwind
   8:        0x10b41ee30 - core::panicking::panic_fmt::h257ceb0aa351d801
   9:        0x10b32a4a9 - core::result::unwrap_failed::h359024e7ee7aee6d
  10:        0x10b3159a7 - cargo_profiler::real_main::hfa05bdcb6b661735
  11:        0x10b313fd3 - cargo_profiler::main::hbb5817797bfb8430
  12:        0x10b3f7acd - std::panicking::try::call::hc5e1f5b484ec7f0e
  13:        0x10b3fb39b - __rust_try
  14:        0x10b3fb335 - __rust_maybe_catch_panic
  15:        0x10b3f78f1 - std::rt::lang_start::h61f4934e780b4dfc
```

It would be nice if it caught these cases and printed something akin to

```
Error: Your Cargo.toml is missing. Are you sure you're in a Rust project?
```

**Comment by kernelmachine** [comment]
Huh, I thought I caught that error. I don't have access to the code base right now - going to look at it later this evening and will circle back. Keep the bugs coming! Thanks :)

**Comment by regexident** [comment]
As long as you keep smashing them. 😋

Joking aside, if you need more samples with different projects, lemme know.

**Comment by regexident** [comment]
Ugh, that second sentence was meant for the regex bug. Mobile, hrmpf… ¯_(ツ)_/¯

**Comment by kernelmachine** [comment]
you got it! This weekend I want to develop a bunch of tests to help prevent these unexpected errors. Unfortunately it's difficult to get complete coverage ahead of time for this type of project, but i think we can build the tests for individual functions that are independent of the actual binary supplied.

**Comment by kernelmachine** [comment]
more samples would be very very useful.

**Comment by kernelmachine** [comment]
This is fixed in v0.1.4
